hbase-client/src/main/java/org/apache/hadoop/hbase/client/BufferedMutatorImpl.java

```diff
@@ -235,6 +253,15 @@ public synchronized void close() throws IOException {
     }
     // Stop any running Periodic Flush timer.
     disableWriteBufferPeriodicFlush();
+    ap.waitAllSlot();
+    try {
+      // Let time to the periodic flush thread to exit (task are finished, but not the code after)
```

Comment:
What is the problem if the code after is not finished? I'm a bit nervous that we just set a magic 5ms sleep time here...